### PR TITLE
Allow startup using a node name when using a config properties file

### PR DESCRIPTION
**Pull request by saurabhagas**
Allow startup using a node name when using a config properties file

### CODE REVIEW

File: dynamic-config/model/src/main/java/org/terracotta/dynamic_config/api/model/Setting.java [hunk 1]
```diff
@@ -83,7 +83,7 @@
       NODE,
       fromNode(Node::getNodeName),
       intoNode(Node::setNodeName),
-      of(GET, CONFIG),
+      of(GET, SET, CONFIG),
```
Comment (mathieucarbou) [inline dynamic-config/model/src/main/java/org/terracotta/dynamic_config/api/model/Setting.java:86]: `SET` is sued to describe a setting that can be changed through config-tool set command. If this is the goal, some list below should probably be updated (i.e. requrements, etc), and some extensive unit tests added because right now, the same + stripeId is used as a couple to identify a node, so this will completely break the detection we have in Configuratio class and DyanmicConfigServiceImpl to be able to determine which node we were referring to.
If the name can be changed, we have also to get rid at ALL he places (including NodeContext) where we can refer to a node by using the stripeId / name couple.

Anyway ;-)
I will let you test and discover all the problematic use cases and consequences this has.
The code as it is right now it really not designed to allow a node name change.

File: dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/startup/Options.java [hunk 2]
```diff
@@ -136,7 +136,7 @@
   @Parameter(names = {"-N", "--" + CLUSTER_NAME})
   private String clusterName;
 
-  @Parameter(names = {"-l", "--" + LICENSE_FILE})
+  @Parameter(names = {"-l", "--" + LICENSE_FILE}, hidden = true)
```
Comment (mathieucarbou) [inline dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/startup/Options.java:139]: should be moved to a separate commit and explain why

File: dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/startup/Options.java [hunk 3]
```diff
@@ -188,53 +187,47 @@ private void extractTopologyOptions(CustomJCommander jCommander) {
...
+      filteredOptions.remove(addDashDash(NODE_NAME));
+      filteredOptions.remove("-n");
       filteredOptions.remove(addDashDash(NODE_CONFIG_DIR));
+      filteredOptions.remove("-r");
 
       if (filteredOptions.size() != 0) {
         throw new ParameterException(
             String.format(
-                "'%s' parameter can only be used with '%s', '%s', '%s', '%s' and '%s' parameters",
+                "'%s' parameter can only be used with '%s', '%s' and '%s' parameters",
```
Comment (mathieucarbou) [inline dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/startup/Options.java:207]: why removing hostname/port support ? We should be able to use -f with hostname/port or with nodeName, as we want.
The idea behind hostname/port is the usage in cloud, where node names could be unknowm.

Comment (saurabhagas) [reply dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/startup/Options.java]: Done

File: dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/startup/Options.java (hunk 2 above)
Comment (saurabhagas) [reply dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/startup/Options.java:139]: The changes were all in the same file - will be hard to fish out the changes for this. The commit message contains an explanation for this though.

File: dynamic-config/model/src/main/java/org/terracotta/dynamic_config/api/model/Setting.java (hunk 1 above)
Comment (saurabhagas) [reply dynamic-config/model/src/main/java/org/terracotta/dynamic_config/api/model/Setting.java]: Reverted, as per discussion over email.
